### PR TITLE
Fixes #4089

### LINKED ISSUE #4089: `deepEquals` returns `true` for different URLs.
### What version of Bun is running?

0.7.4

### What platform is your computer?

Linux 6.4.6-1-cachyos x86_64 unknown

### What steps can reproduce the bug?

```ts
console.log(Bun.deepEquals(new URL("https://bun.sh/"), new URL("https://deno.com/")));
```

### What is the expected behavior?

console should log `false`

### What do you see instead?

console logs `true`

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Fixes #4089 

### How did you verify your code works?

Automated tests

### CODE REVIEW

File: src/bun.js/bindings/bindings.cpp [hunk 1]
```diff
@@ -616,6 +619,24 @@ bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2,
...
+
+                if (auto* url2 = jsDynamicCast<JSDOMURL*>(v2)) {
+                    return url1->wrapped().href() == url2->wrapped().href();
+                }
+
+                if constexpr (isStrict) {
+                    return false;
+                }
+            }
+        }
```
Comment (dylan-conway) [inline src/bun.js/bindings/bindings.cpp:649]: should we return false if one is a url and the other isn't?

Comment (Jarred-Sumner) [reply src/bun.js/bindings/bindings.cpp:649]: maybe just for isStrictEqual?

Comment (dylan-conway) [reply src/bun.js/bindings/bindings.cpp:649]: yeah, that sounds good

**Review by dylan-conway — APPROVED** [review]
LGTM after isStrictEqual change

**Pull request merged**